src/librados/librados_asio.h

```diff
@@ -78,7 +78,7 @@ struct bound_completion_handler : public invoker<Result> {
     // static check for CompletionHandler concept (must be CopyConstructible and
     // callable with no arguments)
     using namespace boost::asio;
-    BOOST_ASIO_COMPLETION_HANDLER_CHECK(bound_completion_handler, *this) type_check;
+    BOOST_ASIO_LEGACY_COMPLETION_HANDLER_CHECK(bound_completion_handler, *this) type_check;
```

Comment:
just leaving a note that this line will go away eventually. it was tagged as 'legacy' because its CopyConstructible requirement was removed. there's work in progress on a generalized `async::Completion` type in https://github.com/ceph/ceph/pull/21271 - once that merges, i plan to use it to replace most of this stuff